packages/nextcloud/test/litmus_test.dart

```diff
@@ -0,0 +1,246 @@
+        expect(response.statusCode, 204);
+
+        for (var i = 0; i < 10; i++) {
+          response = await client.webdav.delete('dst1/$i.txt');
+          expect(response.statusCode, 204);
+        }
+
+        response = await client.webdav.delete('dst1/sub');
+        expect(response.statusCode, 204);
+
+        response = await client.webdav.delete('dst2');
+        expect(response.statusCode, 204);
+      });
+
+      test('copy_shallow', () async {
+        // Does not work on litmus, let's wait for https://github.com/nextcloud/server/issues/39627
+      });
+
+      test('move', () async {
+        await client.webdav.put(Uint8List(0), 'src1.txt');
+        await client.webdav.put(Uint8List(0), 'src2.txt');
+        await client.webdav.mkcol('coll');
+
+        var response = await client.webdav.move('src1.txt', 'dst.txt');
+        expect(response.statusCode, 201);
+
+        expect(
+          () => client.webdav.move('src2.txt', 'dst.txt'),
+          throwsA(predicate((final e) => (e! as DynamiteApiException).statusCode == 412)),
+        );
+
+        response = await client.webdav.move('src2.txt', 'dst.txt', overwrite: true);
+        expect(response.statusCode, 204);
+      });
+
+      test('move_coll', () async {
+        await client.webdav.mkcol('src');
+        await client.webdav.mkcol('src/sub');
+        for (var i = 0; i < 10; i++) {
+          await client.webdav.put(Uint8List(0), 'src/$i.txt');
+        }
+        await client.webdav.put(Uint8List(0), 'noncoll');
+        await client.webdav.copy('src', 'dst2');
+        await client.webdav.move('src', 'dst1');
+
+        expect(
+          () => client.webdav.move('dst1', 'dst2'),
+          throwsA(predicate((final e) => (e! as DynamiteApiException).statusCode == 412)),
+        );
+
+        await client.webdav.move('dst2', 'dst1', overwrite: true);
+        await client.webdav.copy('dst1', 'dst2');
+
+        for (var i = 0; i < 10; i++) {
+          final response = await client.webdav.delete('dst1/$i.txt');
+          expect(response.statusCode, 204);
+        }
+
+        final response = await client.webdav.delete('dst1/sub');
+        expect(response.statusCode, 204);
+
+        expect(
+          () => client.webdav.move('dst2', 'noncoll'),
+          throwsA(predicate((final e) => (e! as DynamiteApiException).statusCode == 412)),
+        );
+      });
+    });
+
+    group('largefile', () {
+      final largefileSize = pow(10, 9).toInt(); // 1GB
+
+      test('large_put', () async {
+        final response = await client.webdav.put(Uint8List(largefileSize), 'test.txt');
+        expect(response.statusCode, 201);
+      });
+
+      test('large_get', () async {
+        final response = await client.webdav.put(Uint8List(largefileSize), 'test.txt');
+        expect(response.statusCode, 201);
+
+        final downloadedContent = await client.webdav.get('test.txt');
+        expect(downloadedContent, hasLength(largefileSize));
+      });
```

Comment:
Do you think it makes sense to unify both tests?

We could have one single test and add the `get` or `put` information in the `reason` field of the `expect`.
I think that's better as a failing put will also result in a failing get thus I don't see a reason to separate them.

Reply:
I was thinking the same. The official litmus tests are not independent of each other so they need this but we don't